src/app/components/harvest/screens/metadata-review/metadata-review.component.ts

```diff
@@ -248,22 +248,29 @@ export class MetadataReviewComponent
       : row.harvestItem?.path;
 
   public updateHarvestWithMappingChange(): void {
-    const mappings: HarvestMapping[] = [];
-
     /*
-     * Iterate through all the rows, extract folders which have mappings and
-     * push that list to the mappings array. We could use a filter and map for
-     * this, however it's likely less performant
-     */
+    * Iterate through all the rows, extract folders which have mappings and
+    * push that list to the mappings array. We could use a filter and map for
+    * this, however it's likely less performant
+    */
     this.rows.forEach((row): void => {
       if (row.rowType === RowType.folder && isInstantiated(row.mapping)) {
-        mappings.push(row.mapping);
+
+        if (this.harvest.mappings.find(mapping => mapping.path === row.mapping.path)) {
+          // since the path attribute is a unique identifier on a mapping, we can safely assume
+          // that there will ever only be one mapping returned from the bellow filter
+          Object.assign(this.harvest.mappings.filter(mapping => mapping.path === row.mapping.path)[0], row.mapping);
+        } else {
+          // the mapping doesn't exist on the current harvest model, so create a new one
+          this.harvest.mappings.push(row.mapping);
+        }
+
       }
     });
 
     this.hasUnsavedChanges = true;
     this.harvestApi
-      .updateMappings(this.harvest, mappings)
+      .updateMappings(this.harvest, this.harvest.mappings)
```

Comment:
i'm usually all for storing information on a model, but in this case it's temporary. 

I think keep the constructed form separate, and then update the harvest mappings after you get the response back from the server (so we have authoritative mappings)

Actually I'm undecided on this. You can keep it on the model, but it should still be updated with the response from the server afterwards